homeassistant/components/atag/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for the Atag component."""
+from aiohttp import ClientSession
+from pyatag import DEFAULT_PORT, AtagDataStore, AtagException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_HOST, CONF_PORT
+from homeassistant.core import callback
+
+from . import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = {
+    vol.Required(CONF_HOST): str,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+}
+
+
+class AtagConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Atag."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        if not user_input:
+            return await self._show_form()
+
+        try:
+            async with ClientSession() as session:
```

Comment:
Why do we create a new session here? Can we not use the home assistant session?

Reply:
I'm not an expert on sessions tbh, but I ran into closed client session if I used the HA session in the config flow and later again in the setup flow. Any idea why that would happen? In any case I'll try with the HA session again

Reply:
Does the library close the session?